va/va.c

```diff
@@ -529,19 +527,20 @@ static VAStatus va_openDriver(VADisplay dpy, char *driver_name)
     }
     driver_dir = strtok_r(search_path, ENV_VAR_SEPARATOR, &saveptr);
     while (driver_dir) {
-        char *driver_path = va_getDriverPath(driver_dir, driver_name);
-        if (!driver_path) {
-            va_errorMessage(dpy, "%s L%d Out of memory\n",
-                            __FUNCTION__, __LINE__);
-            free(search_path);
-            return VA_STATUS_ERROR_ALLOCATION_FAILED;
-        }
-
-        vaStatus = va_openDriverFromPath(dpy, driver_path);
-        free(driver_path);
-        if (VA_STATUS_SUCCESS == vaStatus)
-            break;
+        for (int use_suffix = 1; use_suffix >= 0; use_suffix--) {
+            char *driver_path = va_getDriverPath(driver_dir, driver_name, use_suffix);
```

Comment:
will this construct `vaon12_drv_video_drv_video` with the changed default driver name for windows?

Reply:
Yes, the first attempt will be `vaon12_drv_video_drv_video` and then fallback to `vaon12_drv_video` (no added suffix). This is done like this (and in this order) to prevent any functional changes that may break other backends that follow the strict name suffix on Linux. While this adds an extra attempt to the loading on Windows, I think it's less invasive than trying without suffix first (that would add an extra attempt to all other backends before finding their driver files on Linux)